website/docs/tutorials/07-testing/02-cheatcodes/given.md

```diff
@@ -0,0 +1,42 @@
+# `given`
+
+```python
+def given(**kwargs: StrategyDescriptor) -> None:
+```
+
+Instructs the fuzzer to adopt a new fuzzing strategy to input parameters.
+The built-in strategies are provided by the [`strategy`](./strategy.md) cheatcode, and a list of
+available strategies, see the [fuzzing strategies](../03-fuzzing/strategies.md) guide page.
+
+:::warning
+This cheatcode is only available in [fuzz tests](../fuzzing).
+:::
+
+```cairo title="Example"
+@external
+func test_integers{syscall_ptr : felt*, range_check_ptr}(a : felt, b : felt):
+    %{
+        given(
+            a = strategy.integers(10, 20),
+            b = strategy.integers(30, 40),
+        )
+    %}
+    assert_le(a, b)
+    return ()
+end
+```
+
+## Performance caveats
+
+While it is possible to call `given` multiple times in a test case, try to always do it once.
+Multiple input parameters should be configured by passing many arguments to the cheatcode.
+Due to Cairo language limitations and resulting therefrom implementation details,
+each call to the `given` cheatcode causes the fuzzer to restart immediately with a fresh state.
```

Comment:
Is the information about fuzzer restarting necessary here 🤔

Reply:
It is, because this has many consequences:
1. Random seed will be restarted.
2. Fuzzer will explore the same values once again.
3. If we add #499 then it may happen that fuzzer will test these examples again also

Reply:
Maybe these consequences should be in the doc.

Reply:
Will user know about this test being performed? Or It will be under the hood?

Reply:
User will know. These tests subtract max_examples, so that sum of all examples over all learning steps is = max_examples (sans shrinking phase)